libdnf5-cli/output/transactionlist.cpp

```diff
@@ -49,11 +51,15 @@ void print_transaction_list(std::vector<libdnf5::transaction::Transaction> & ts_
         scols_line_set_data(ln, 2, libdnf5::utils::string::format_epoch(ts.get_dt_start()).c_str());
         // TODO(lukash) fill the Actions(s), if we even want them?
         scols_line_set_data(ln, 3, "");
-        scols_line_set_data(
-            ln,
-            4,
-            std::to_string(ts.get_packages().size() + ts.get_comps_groups().size() + ts.get_comps_environments().size())
-                .c_str());
+        std::string trans_item_count;
+        if (id_to_item_count.at(ts.get_id())) {
+            trans_item_count = std::to_string(id_to_item_count.at(ts.get_id()));
+        } else {
+            // Fallback to manual calculation (very slow if the packages, groups and environments have to be fetched from DB)
```

Comment:
Is it even necessary to have a fallback here? Couldn't we just call `TransactionHistory::get_transaction_item_counts` before printing the transaction list here and that's it?

Reply:
Updated.